core/drivers/crypto/se050/adaptors/apis/user.c

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <nxScp03_Apis.h>
+#include <se050_user_apis.h>
+#include <se050_utils.h>
+#include <string.h>
+
+/*
+ * @param k_object
+ * @param k_store
+ * @param key_id
+ *
+ * @return sss_status_t
+ */
+static sss_status_t alloc_scp_key_to_auth(sss_object_t *k_object,
+					  sss_key_store_t *k_store,
+					  uint32_t key_id)
+{
+	sss_status_t status = kStatus_SSS_Fail;
+
+	if (!k_object || !k_store)
+		return kStatus_SSS_Fail;
+
+	status = sss_host_key_object_init(k_object, k_store);
+	if (status != kStatus_SSS_Success)
+		return status;
+
+	status = sss_host_key_object_allocate_handle(k_object, key_id,
+						     kSSS_KeyPart_Default,
+						     kSSS_CipherType_AES, 16,
+						     kKeyObject_Mode_Transient);
+	return status;
+}
+
+/*
+ * @param scp
+ * @param auth
+ * @param k_store
+ * @param oid
+ *
+ * @return sss_status_t
+ */
+static sss_status_t prepare_host_scp(NXSCP03_AuthCtx_t *scp,
+				     struct se050_auth_ctx *auth,
+				     sss_key_store_t *k_store,
+				     struct se050_scp_key *keys,
+				     uint32_t oid)
+{
+	sss_status_t status = kStatus_SSS_Fail;
+	NXSCP03_StaticCtx_t *pStatic_ctx = NULL;
+	NXSCP03_DynCtx_t *pDyn_ctx = NULL;
+	size_t len = 0;
+
+	if (!scp || !auth || !k_store)
+		return kStatus_SSS_Fail;
+
+	pStatic_ctx = &auth->static_ctx;
+	pDyn_ctx = &auth->dynamic_ctx;
+
+	scp->pStatic_ctx = pStatic_ctx;
+	scp->pDyn_ctx = pDyn_ctx;
+	pStatic_ctx->keyVerNo = 0x0B;
+
+	/* ENC Static Key */
+	status = alloc_scp_key_to_auth(&pStatic_ctx->Enc, k_store, oid++);
+	if (status != kStatus_SSS_Success)
```

Comment:
If the value of `kStatus_SSS_Success` is 0, a simple `if (status)` is preferred.

Reply:
unfortunately is 0x5a5a5a5au

Reply:
:open_mouth: ok, then the code is fine as-is!